component/componenttest/nop_factories.go

```diff
@@ -39,7 +40,8 @@ func NopFactories() (component.Factories, error) {
 		return component.Factories{}, err
 	}
 
-	if factories.Processors, err = component.MakeProcessorFactoryMap(NewNopProcessorFactory()); err != nil {
+	//nolint:staticcheck
```

Comment:
Do you need this here but not in receiver? I am a bit confused.

Reply:
Seems it's not needed on extensions or exporters lines either. I've removed for the others as well.